### PR TITLE
[WICKED] Refactor advanced suit

### PR DESCRIPTION
**Pull request by jlausuch**
- Related ticket: https://progress.opensuse.org/issues/42641
- Verification run: http://fromm.arch.suse.de/tests/2158

### CODE REVIEW

File: lib/main_common.pm [hunk 1]
```diff
@@ -1654,8 +1654,34 @@ sub load_wicked_tests {
         loadtest 'wicked/config_files';
     }
     elsif (check_var('WICKED', 'advanced')) {
-        loadtest 'wicked/advanced_ref' if check_var('IS_WICKED_REF', '1');
-        loadtest 'wicked/advanced_sut' if check_var('IS_WICKED_REF', '0');
+        if (check_var('IS_WICKED_REF', '1')) {
+            loadtest 'wicked/advanced/ref/test01';
```
Comment (asmorodskyi) [inline lib/main_common.pm]: maybe something more generic ? 
```
for (my $i=1; $i <= 12; $i++) {
loadtest 'wicked/advanced/ref/test' . $i < 10 ? '0' . $i : $i ; 
}
```

Comment (cfconrad) [reply lib/main_common.pm]: sprintf("%02d", $i) for the leading 0

File: lib/main_common.pm [hunk 2]
```diff
@@ -1654,8 +1654,34 @@ sub load_wicked_tests {
...
+            loadtest 'wicked/advanced/ref/test05';
+            loadtest 'wicked/advanced/ref/test06';
+            loadtest 'wicked/advanced/ref/test07';
+            loadtest 'wicked/advanced/ref/test08';
+            loadtest 'wicked/advanced/ref/test09';
+            loadtest 'wicked/advanced/ref/test10';
+            loadtest 'wicked/advanced/ref/test11';
+            loadtest 'wicked/advanced/ref/test12';
+        }
+        if (check_var('IS_WICKED_REF', '0')) {
```
Comment (asmorodskyi) [inline lib/main_common.pm]: ... even MORE generic :) 

```
for (my $i=1; $i <= 12; $i++) {
loadtest 'wicked/advanced/' . get_var('IS_WICKED_REF') ? 'ref' : 'sut' . '/test' . $i < 10 ? '0' . $i : $i ; 
}
```

File: tests/wicked/advanced/ref/test01.pm [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+# Copyright © 2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Advanced test cases for wicked
+# Test 1 : Create a GRE interface from legacy ifcfg files
+# Maintainer: Anton Smorodskyi <asmorodskyi@suse.com>, Jose Lausuch <jalausuch@suse.com>
```
Comment (asmorodskyi) [inline tests/wicked/advanced/ref/test01.pm]: don't give any chance @cfconrad  to run away :) he is hardlinked to those two guys :)

File: tests/wicked/advanced/ref/test02.pm [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+#
+# Copyright © 2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Advanced test cases for wicked
+# Test 1 : Create a GRE interface from legacy ifcfg files
```
Comment (asmorodskyi) [inline tests/wicked/advanced/ref/test02.pm]: copy/paste trap

Comment (jlausuch) [reply tests/wicked/advanced/ref/test02.pm]: oops :)

File: tests/wicked/advanced/ref/test01.pm (hunk 3 above)
Comment (jlausuch) [reply tests/wicked/advanced/ref/test01.pm]: sure, I just copy pasted it :)

File: lib/main_common.pm (hunk 1 above)
Comment (jlausuch) [reply lib/main_common.pm]: much better, thanks

File: lib/main_common.pm [hunk 5]
```diff
@@ -1654,8 +1654,34 @@ sub load_wicked_tests {
...
+            loadtest 'wicked/advanced/ref/ipip_tunnel_legacy';
+            loadtest 'wicked/advanced/ref/ipip_tunnel_xml';
+            loadtest 'wicked/advanced/ref/sit_tunnel_legacy';
+            loadtest 'wicked/advanced/ref/sit_tunnel_xml';
+            loadtest 'wicked/advanced/ref/tap_interface_legacy';
+            loadtest 'wicked/advanced/ref/tap_interface_xml';
+            loadtest 'wicked/advanced/ref/tun_interface_legacy';
+            loadtest 'wicked/advanced/ref/tun_interface_xml';
+        }
+        if (check_var('IS_WICKED_REF', '0')) {
```
Comment (cfconrad) [inline lib/main_common.pm]: like @asmorodskyi mentioned but now without the number...
```
my $f = get_var('IS_WICKED_REF') ? 'ref' : 'sut';
loadtest 'wicked/advanced/' . $f . '/bridge_interface_xml';
...
```
We only need to write each testmodule once...

File: lib/main_common.pm [hunk 6]
```diff
@@ -1654,8 +1654,34 @@ sub load_wicked_tests {
         loadtest 'wicked/config_files';
     }
     elsif (check_var('WICKED', 'advanced')) {
-        loadtest 'wicked/advanced_ref' if check_var('IS_WICKED_REF', '1');
-        loadtest 'wicked/advanced_sut' if check_var('IS_WICKED_REF', '0');
+        if (check_var('IS_WICKED_REF', '1')) {
+            loadtest 'wicked/advanced/ref/bridge_interface_legacy';
```
Comment (cfconrad) [inline lib/main_common.pm]: should we prefixing the filename with the testnumber?

File: tests/wicked/advanced/ref/tun_interface_legacy.pm [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+use mmapi;
+
+sub run {
+    my ($self)         = @_;
+    my $config         = '/etc/sysconfig/network/ifcfg-tun1';
+    my $openvpn_server = '/etc/openvpn/server.conf';
+    record_info('Info', 'Create a TUN interface from legacy ifcfg files');
+    $self->get_from_data('wicked/ifcfg/tun1_ref',      $config);
+    $self->get_from_data('wicked/openvpn/server.conf', $openvpn_server);
+    assert_script_run("sed \'s/device/tun1/\' -i $openvpn_server");
```
Comment (cfconrad) [inline tests/wicked/advanced/ref/tun_interface_legacy.pm]: The escape of single quote is not necessary.

File: tests/wicked/advanced/ref/tun_interface_xml.pm [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+use mmapi;
+
+sub run {
+    my ($self)         = @_;
+    my $config         = '/etc/sysconfig/network/ifcfg-tun1';
+    my $openvpn_server = '/etc/openvpn/server.conf';
+    record_info('Info', 'Create a TUN interface from Wicked XML files');
+    $self->get_from_data('wicked/ifcfg/tun1_ref',      $config);
+    $self->get_from_data('wicked/openvpn/server.conf', $openvpn_server);
+    assert_script_run("sed \'s/device/tun1/\' -i $openvpn_server");
```
Comment (cfconrad) [inline tests/wicked/advanced/ref/tun_interface_xml.pm]: Same for `\'`

File: tests/wicked/advanced/sut/bridge_interface_legacy.pm [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+
+sub run {
+    my ($self) = @_;
+    my $config = '/etc/sysconfig/network/ifcfg-br0';
+    my $dummy  = '/etc/sysconfig/network/ifcfg-dummy0';
+    record_info('Info', 'Create Bridge interface from legacy ifcfg files');
+    $self->get_from_data('wicked/ifcfg/br0',    $config);
+    $self->get_from_data('wicked/ifcfg/dummy0', $dummy);
+    $self->setup_bridge($config, $dummy, 'ifup');
+    my $res = $self->get_test_result("br0", "");
```
Comment (cfconrad) [inline tests/wicked/advanced/sut/bridge_interface_legacy.pm]: Second parameter of get_test_result() isn't necessary. If you going to change it, the other files have it too.
Maybe change to  single quote.

File: lib/main_common.pm [hunk 10]
```diff
@@ -1654,8 +1654,19 @@ sub load_wicked_tests {
         loadtest 'wicked/config_files';
     }
     elsif (check_var('WICKED', 'advanced')) {
-        loadtest 'wicked/advanced_ref' if check_var('IS_WICKED_REF', '1');
-        loadtest 'wicked/advanced_sut' if check_var('IS_WICKED_REF', '0');
+        my $f = get_var('IS_WICKED_REF') ? 'ref' : 'sut';
```
Comment (cfconrad) [inline lib/main_common.pm]: sorry I was a bit late with my change.
Plz use `check_var('IS_WICKED_REF', '1');` I just saw that IS_WICKED_REF is set for both

Comment (cfconrad) [reply lib/main_common.pm]: Ah no it will work in both ways.

Comment (asmorodskyi) [reply lib/main_common.pm]: as far as I know perl will treat '0' as false , no ?

**Pull request merged**